zerver/views/streams.py

```diff
@@ -77,31 +78,44 @@ def add_default_stream(request, user_profile, stream_name=REQ()):
 @require_realm_admin
 @has_request_variables
 def create_default_stream_group(request: HttpRequest, user_profile: UserProfile,
-                                group_name: Text=REQ(),
+                                group_name: Text=REQ(), description: Text=REQ(),
                                 stream_names: List[Text]=REQ(validator=check_list(check_string))) -> None:
     streams = []
     for stream_name in stream_names:
         (stream, recipient, sub) = access_stream_by_name(user_profile, stream_name)
         streams.append(stream)
-    do_create_default_stream_group(user_profile.realm, group_name, streams)
+    do_create_default_stream_group(user_profile.realm, group_name, description, streams)
     return json_success()
 
 @require_realm_admin
 @has_request_variables
 def update_default_stream_group(request: HttpRequest, user_profile: UserProfile,
                                 group_name: Text=REQ(), op: Text=REQ(),
-                                stream_names: List[Text]=REQ(validator=check_list(check_string))) -> None:
+                                new_group_name: Text=REQ(validator=check_string, default=None),
+                                new_description: Text=REQ(validator=check_string, default=None),
+                                stream_names: List[Text]=REQ(validator=check_list(check_string), default=[])) -> None:
     streams = []
     for stream_name in stream_names:
         (stream, recipient, sub) = access_stream_by_name(user_profile, stream_name)
         streams.append(stream)
 
     if op == 'add':
+        if not streams:
+            return json_error(_('Specify stream names to add to "%s".') % (group_name,))
         do_add_streams_to_default_stream_group(user_profile.realm, group_name, streams)
     elif op == 'remove':
+        if not streams:
+            return json_error(_('Specify stream names to remove from "%s".') % (group_name,))
         do_remove_streams_from_default_stream_group(user_profile.realm, group_name, streams)
+    elif op == 'change':
+        if new_group_name is not None:
+            do_change_default_stream_group_name(user_profile.realm, group_name, new_group_name)
+        if new_description is not None:
+            do_change_default_stream_group_description(user_profile.realm, group_name, new_description)
+        if not new_group_name and not new_description:
+            return json_error(_('You must pass "new_description" or "new_group_name".'))
     else:
-        return json_error(_('Nothing to do. Specify at least one of "add" or "remove".'))
+        return json_error(_('Nothing to do. Specify one of "add", "remove" or "change".'))
```

Comment:
This structure is kinda a mess.  I think we should just use separate endpoints for modifying members vs. changing the group's properties.  I think `PATCH /default_stream_groups/(?P<name>)/streams` and `PATCH /default_stream_groups/(?P<name>)` would be pretty reasonable endpoints for these two things.